services/api/service.go

```diff
@@ -1307,16 +1294,27 @@ func (api *RelayAPI) handleSubmitNewBlock(w http.ResponseWriter, req *http.Reque
 	}
 
 	// Verify the signature
-	log = log.WithField("timestampBeforeSignatureCheck", time.Now().UnixMilli())
+	log = log.WithField("timestampBeforeSignatureCheck", time.Now().UTC().UnixMilli())
 	builderPubkey := payload.BuilderPubkey()
 	signature := payload.Signature()
 	ok, err = boostTypes.VerifySignature(payload.Message(), api.opts.EthNetDetails.DomainBuilder, builderPubkey[:], signature[:])
+	log = log.WithField("timestampAfterSignatureCheck", time.Now().UTC().UnixMilli())
 	if !ok || err != nil {
 		log.WithError(err).Warn("could not verify builder signature")
 		api.RespondError(w, http.StatusBadRequest, "invalid signature")
 		return
 	}
 
+	// Reject new submissions once the payload for this slot was delivered - TODO: store in memory as well
+	slotLastPayloadDelivered, err := api.redis.GetStatsUint64(datastore.RedisStatsFieldSlotLastPayloadDelivered)
+	if err != nil && !errors.Is(err, redis.Nil) {
+		log.WithError(err).Error("failed to get delivered payload slot from redis")
+	} else if payload.Slot() <= slotLastPayloadDelivered {
```

Comment:
nit: the way the current conditional is structured, we would reach the `else if` clause when `err != nil` and `errors.Is(err, redis.Nil` 

That would cause us to perform a comparison with an invalid `slotLastPayloadDelivered` 

Since the value returned on error is 0, it probably has not caused issue, but it would be nice if we can make the logic clearer and avoid unnecessary comparison

Reply:
great catch